nexus/src/external_api/http_entrypoints.rs

```diff
@@ -4023,6 +4024,97 @@ async fn updates_refresh(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+// Metrics
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct ResourceUtilization {
+    #[serde(flatten)]
+    pub pagination: dropshot::PaginationParams<
+        params::ResourceMetrics,
+        params::ResourceMetrics,
+    >,
+
+    /// The UUID of the container being queried
+    // TODO: I might want to force the caller to specify type here?
+    pub id: Uuid,
+}
+
+#[derive(Display, Deserialize, JsonSchema)]
+#[display(style = "snake_case")]
+#[serde(rename_all = "snake_case")]
+pub enum ResourceName {
```

Comment:
```suggestion
pub enum MetricName {
```

?  (I don't get the sense in which these are names of resources, except maybe HTTP resources, but we don't usually use "resource" that way in the API and documentation)

Reply:
I think `MetricName` was already taken (could probably use a rename too). But `SystemMetricName` would be fine. I agree resource is misleading.

Reply:
Done in 91a1b9dc7c5e4fbf047b4c548f2c19f5f9f8f4b4